src/github/SyftGithubAction.ts

```diff
@@ -0,0 +1,398 @@
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as github from "@actions/github";
+import * as cache from "@actions/tool-cache";
+import {
+  PullRequestEvent,
+  Release,
+  ReleaseEvent,
+} from "@octokit/webhooks-types";
+import * as fs from "fs";
+import * as os from "os";
+import path from "path";
+import stream from "stream";
+import { SyftOptions } from "../Syft";
+import { dashWrap, getClient } from "./GithubClient";
+
+export const SYFT_BINARY_NAME = "syft";
+export const SYFT_VERSION = "v0.21.0";
+
+const PRIOR_ARTIFACT_ENV_VAR = "ANCHORE_SBOM_ACTION_PRIOR_ARTIFACT";
+
+/**
+ * Tries to get a unique artifact name or otherwise as appropriate as possible
+ */
+function getArtifactName(): string {
+  const fileName = core.getInput("artifact_name");
+
+  // if there is an explicit filename just return it, this could cause issues
+  // where earlier sboms are overwritten by later ones
+  if (fileName) {
+    return fileName;
+  }
+
+  const { job, action } = github.context;
+  // when run without an id, we get various auto-generated names, like:
+  // __self __self_2 __anchore_sbom-action  __anchore_sbom-action_2 etc.
+  // so just keep the number at the end if there is one, otherwise
+  // this will not match an id unless for some reason it starts with __
+  let stepName = action.replace(/__[-_a-z]+/, "");
+  if (stepName) {
+    stepName = `-${stepName}`;
+  }
+  const format = getSbomFormat();
+  return `sbom-${job}${stepName}.${format}`;
+}
+
+/**
+ * Gets a reference to the syft command and executes the syft action
+ * @param input syft input parameters
+ * @param format syft output format
+ */
+async function executeSyft({ input, format }: SyftOptions): Promise<string> {
+  let stdout = "";
+  let stderr = "";
+
+  const cmd = await getSyftCommand();
+
+  const env: { [key: string]: string } = {
+    SYFT_CHECK_FOR_APP_UPDATE: "false",
+  };
+
+  // https://github.com/anchore/syft#configuration
+  let args = ["packages"];
+
+  if ("image" in input && input.image) {
+    args = [...args, `docker:${input.image}`];
+  } else if ("path" in input && input.path) {
+    args = [...args, `dir:${input.path}`];
+  } else {
+    throw new Error("Invalid input, no image or path specified");
+  }
+
+  args = [...args, "-o", format];
+
+  let error: unknown;
+  try {
+    // Execute in a group so the syft output is collapsed in the GitHub log
+    core.info(`[command]${cmd} ${args.join(" ")}`);
+
+    // Need to implement this /dev/null writable stream so the entire contents
```

Comment:
Should we update this to a TODO? I'm having trouble following where the buffer comes from in this case. Also my IDE is saying
```
'buffer' is declared but its value is never read.
```

Reply:
I looked at the demo of this being run and it doesn't looks like the SBOM is being written in the log. Basically do we still need this comment? What changes have to be made? Does it only print in certain cases?

Reply:
the _reason_ the SBOM is not being written to the log is not written is because of this "/dev/null" stream. the comment must not have been clear enough.

re: buffer never read, i need the 3rd parameter... I guess i could name them `_`

Reply:
... clarified the comment in the code

Reply:
FWIW Idea recognizes that I need the 3rd parameter: `next` and doesn't complain about unused variables... I'm going to leave this for now